src/lib.rs

```diff
@@ -161,6 +161,17 @@ impl MachineSetup {
     }
 }
 
+#[derive(Default)]
+struct SpotRequestResult<'a> {
```

Comment:
Can you also derive `Debug` and `Clone` here? It's not currently necessary, but those are nice to have :)

Reply:
I can't derive debug because the `Fn` doesn't implement it for some reason. I can implement `Clone`, though

Reply:
Ohh, right, that's too bad. I suppose we could implement `Debug` manually on it, but that should be dealt with in a separate PR if at all :)